include/sound/sof/topology.h

```diff
@@ -163,39 +165,24 @@ struct sof_ipc_comp_tone {
 	int32_t ramp_step;
 } __packed;
 
-/** \brief Types of EFFECT */
-enum sof_ipc_effect_type {
-	SOF_EFFECT_NONE = 0,		/**< None */
-	SOF_EFFECT_INTEL_EQFIR,		/**< Intel FIR */
-	SOF_EFFECT_INTEL_EQIIR,		/**< Intel IIR */
+/** \brief Types of processing components */
+enum sof_ipc_process_type {
+	SOF_PROCESS_NONE = 0,		/**< None */
+	SOF_PROCESS_EQFIR,		/**< Intel FIR */
+	SOF_PROCESS_EQIIR,		/**< Intel IIR */
+	SOF_PROCESS_KEYWORD_DETECT,	/**< Keyword Detection */
+	SOF_PROCESS_VOICE_DETECT,	/**< Voice Detection */
```

Comment:
This is all over the place, I would like this to be broken in parts that can be reviewed independently
1. EQ related changes - and handling of backwards compatibility if needed
2. introduction of keyword detect and voice detect.

Reply:
@plbossart fixed now. 
@singalsu @juimonen this PR is the kernel counterpart for https://github.com/thesofproject/sof/pull/1000. Could I please request you to give these a try and give your input?